### PR TITLE
 feature (account/handers): ldap account and managed group CRUDL APIs

### PR DESCRIPTION
**Pull request by jimlambrt**

### CODE REVIEW

**Review by johanbrandhorst — APPROVED** [review]
LGTM

File: internal/proto/controller/storage/auth/ldap/store/v1/ldap.proto [hunk 1]
```diff
@@ -563,5 +563,25 @@ message ManagedGroup {
 
   // groups is json marshalled list of groups that make up the ManagedGroup
   // @inject_tag: `gorm:"not_null"`
-  string group_names = 80;
+  string group_names = 80 [(custom_options.v1.mask_mapping) = {
+    this: "GroupNames"
+    that: "attributes.group_names"
+  }];
+
```
Comment (irenarindos) [inline internal/proto/controller/storage/auth/ldap/store/v1/ldap.proto:570]:
```suggestion
```
based on: https://app.circleci.com/pipelines/github/hashicorp/boundary/14196/workflows/a9be4de8-f0b8-4438-8743-3ab555ec2f1a/jobs/166150

File: internal/daemon/controller/handlers/managed_groups/managed_group_service.go [hunk 2]
```diff
@@ -326,7 +367,7 @@ func (s Service) createOidcInRepo(ctx context.Context, am auth.AuthMethod, item
 	attrs := item.GetOidcManagedGroupAttributes()
 	mg, err := oidc.NewManagedGroup(ctx, am.GetPublicId(), attrs.GetFilter(), opts...)
 	if err != nil {
-		return nil, handlers.ApiErrorWithCodeAndMessage(codes.Internal, "Unable to build user for creation: %v.", err)
+		return nil, handlers.ApiErrorWithCodeAndMessage(codes.Internal, "Unable to build managed for creation: %v.", err)
```
Comment (irenarindos) [inline internal/daemon/controller/handlers/managed_groups/managed_group_service.go:370]:
```suggestion
		return nil, handlers.ApiErrorWithCodeAndMessage(codes.Internal, "Unable to build managed group for creation: %v.", err)
```

File: internal/proto/controller/storage/auth/ldap/store/v1/ldap.proto (hunk 1 above)
Comment (jimlambrt) [reply internal/proto/controller/storage/auth/ldap/store/v1/ldap.proto:570]: ty!

File: internal/daemon/controller/handlers/managed_groups/managed_group_service.go (hunk 2 above)
Comment (jimlambrt) [reply internal/daemon/controller/handlers/managed_groups/managed_group_service.go]: ty!

**Review by irenarindos — APPROVED** [review]

**Pull request merged**